dipy/segment/metricspeed.pyx

```diff
@@ -0,0 +1,372 @@
+            2*      \
+                c    *2
+
+    is equal to $a+b+c$ where $a$ is the Euclidean distance between s1[0] and
+    s2[0], $b$ between s1[1] and s2[1] and $c$ between s1[2] and s2[2].
+    """
+    cdef double c_dist(SumPointwiseEuclideanMetric self, Data2D features1, Data2D features2) nogil:
+        cdef :
+            int N = features1.shape[0], D = features1.shape[1]
+            int n, d
+            double dd, dist_n, dist = 0.0
+
+        for n in range(N):
+            dist_n = 0.0
+            for d in range(D):
+                dd = features1[n, d] - features2[n, d]
+                dist_n += dd*dd
+
+            dist += sqrt(dist_n)
+
+        return dist
+
+    cdef int c_are_compatible(SumPointwiseEuclideanMetric self, Shape shape1, Shape shape2) nogil:
+        return same_shape(shape1, shape2)
+
+
+cdef class AveragePointwiseEuclideanMetric(SumPointwiseEuclideanMetric):
+    r""" Computes the average of pointwise Euclidean distances between two
+    sequential data.
+
+    A sequence of N-dimensional points is represented as a 2D array with
+    shape (nb_points, nb_dimensions). A `feature` object can be specified
+    in order to calculate the distance between the features, rather than
+    directly between the sequential data.
+
+    Parameters
+    ----------
+    feature : `Feature` object, optional
+        It is used to extract features before computing the distance.
+
+    Notes
+    -----
+    The distance between two 2D sequential data::
+
+        s1       s2
+
+        0*   a    *0
+          \       |
+           \      |
+           1*     |
+            |  b  *1
+            |      \
+            2*      \
+                c    *2
+
+    is equal to $(a+b+c)/3$ where $a$ is the Euclidean distance between s1[0] and
+    s2[0], $b$ between s1[1] and s2[1] and $c$ between s1[2] and s2[2].
+    """
+    cdef double c_dist(AveragePointwiseEuclideanMetric self, Data2D features1, Data2D features2) nogil:
+        cdef int N = features1.shape[0]
+        cdef double dist = SumPointwiseEuclideanMetric.c_dist(self, features1, features2)
+        return dist / N
+
+
+cdef class MinimumAverageDirectFlipMetric(AveragePointwiseEuclideanMetric):
+    r""" Computes the ADF distance (minimum average direct-flip) between two
+    sequential data.
+
+    A sequence of N-dimensional points is represented as a 2D array with
+    shape (nb_points, nb_dimensions).
+
+    Notes
+    -----
+    The distance between two 2D sequential data::
+
+        s1       s2
+
+        0*   a    *0
+          \       |
+           \      |
+           1*     |
+            |  b  *1
+            |      \
+            2*      \
+                c    *2
+
+    is equal to $\min((a+b+c)/3, (a'+b'+c')/3)$ where $a$ is the Euclidean distance
+    between s1[0] and s2[0], $b$ between s1[1] and s2[1], $c$ between s1[2]
+    and s2[2], $a'$ between s1[0] and s2[2], $b'$ between s1[1] and s2[1]
+    and $c'$ between s1[2] and s2[0].
```

Comment:
You forgot to say in the docstring that this is the direct and flipped version of AFD.